Chore/dsys 913/rename popover (#2914)
sumup-oss/circuit-ui: packages/circuit-ui/components/ActionMenu/ActionMenu.mdx

Reviewer: I just realized that this is the existing documentation for the Popover component 😬 

It would be wonderful if you could take the opportunity to improve it, but this isn't a blocker for the pull request.

import { Meta, Status, Props, Story } from '../../../../.storybook/components';
import * as Stories from './ActionMenu.stories';

<Meta of={Stories} />

# ActionMenu

An ActionMenu displays a list of subsequent actions, when interacting with an actionable component.

<Story of={Stories.Base} />
<Props />

> - The reference element, which triggers ActionMenu, can be primary, secondary, tertiary buttons, an overflow icon, or components with embedded buttons such as the [ImageInput](Forms/ImageInput) component.

Reviewer: "overflow icon" makes it sound like an SVG is sufficient as a trigger element (when it should be wrapped in an IconButton) and the ImageInput isn't any different from any other composite component containing a button. I would simplify this to:

```suggestion
- The reference element which triggers the ActionMenu must be a `button` element or a component that renders a `button` such as the Button or IconButton components.
```

> - Each ActionMenu action item is represented by an appropriate HTML element (e.g., a button element or an anchor element).

Reviewer: Rendering the appropriate semantic element is handled internally. What's relevant to developers is that they should include `href` and/or `onClick` props.

- If needed, the dividing line can be used to separate ActionMenu action items.
- The leading icon is optional.

> - The ActionMenu is powered by [Floating UI](https://floating-ui.com/docs/react-dom). You can easily change the position of the ActionMenu relative to the reference element by passing in the `placement` prop. If you want to offset the ActionMenu in the x and y directions, use the `offset` prop.

Reviewer: It's good practice to avoid subjective language such as "easy" or "simple" in documentation. Sarah Rainsberger explains this well: https://www.rainsberger.ca/blog/50-25-just-follow-this-tip/

```suggestion
- The ActionMenu is powered by [Floating UI](https://floating-ui.com/docs/react-dom). You can change the position of the ActionMenu relative to the reference element by passing in the `placement` prop. If you want to offset the ActionMenu in the x and y directions, use the `offset` prop.
```

<Story of={Stories.Offset} />

## Related components

> This component is built on top of the low level [Popover](Components/Popover/Docs) component. If this component does not meet your requirements, you can use the Dialog component directly to build your own custom popover component.

Reviewer: ```suggestion
This component is built on top of the low level [Popover](Components/Popover/Docs) component. If this component does not meet your requirements, you can use the Popover component directly to build your own custom popover component.
```

Author: Actually, I am considering not to rebuild this using the Popover, but keep it based on the Dialog. There isn't really any major gain in terms of complexity, on the contrary, it will lead to some avoidable style overriding.

## Usage guidelines

- **Do** use clear, concise and actionable labels for ActionMenu items

> - **Do** always think about the priority of the action option to be taken and put the option order in logical order

Reviewer: ```suggestion
- **Do** always think about the priority of the actions to be taken and put the options in logical order

```
